src/topology/locally_constant/basic.lean

```diff
@@ -0,0 +1,260 @@
+  is_locally_constant (f / g) :=
+begin
+  rw iff_exists_open at hf hg ⊢,
+  intro x,
+  obtain ⟨U, hU, hxU, HU⟩ := hf x,
+  obtain ⟨V, hV, hxV, HV⟩ := hg x,
+  use [U ∩ V, is_open_inter hU hV, ⟨hxU, hxV⟩],
+  rintro x' ⟨hx'U, hx'V⟩,
+  simp only [pi.div_apply, HU x' hx'U, HV x' hx'V]
+end
+
+end is_locally_constant
+
+/-- A (bundled) locally constant function from a topological space `X` to a type `Y`. -/
+structure locally_constant (X Y : Type*) [topological_space X] :=
+(to_fun : X → Y)
+(is_locally_constant : is_locally_constant to_fun)
+
+namespace locally_constant
+
+instance [inhabited Y] : inhabited (locally_constant X Y) :=
+⟨⟨_, is_locally_constant.const (default Y)⟩⟩
+
+instance : has_coe_to_fun (locally_constant X Y) := ⟨_, locally_constant.to_fun⟩
+
+initialize_simps_projections locally_constant (to_fun → apply)
+
+@[simp] lemma to_fun_eq_coe (f : locally_constant X Y) : f.to_fun = f := rfl
+
+@[simp] lemma coe_mk (f : X → Y) (h) : ⇑(⟨f, h⟩ : locally_constant X Y) = f := rfl
+
+theorem congr_fun {f g : locally_constant X Y} (h : f = g) (x : X) : f x = g x :=
+congr_arg (λ h : locally_constant X Y, h x) h
+
+theorem congr_arg (f : locally_constant X Y) {x y : X} (h : x = y) : f x = f y :=
+congr_arg (λ x : X, f x) h
+
+theorem coe_inj ⦃f g : locally_constant X Y⦄ (h : (f : X → Y) = g) : f = g :=
+by cases f; cases g; cases h; refl
+
+@[ext] theorem ext ⦃f g : locally_constant X Y⦄ (h : ∀ x, f x = g x) : f = g :=
+coe_inj (funext h)
+
+theorem ext_iff {f g : locally_constant X Y} : f = g ↔ ∀ x, f x = g x :=
+⟨λ h x, h ▸ rfl, λ h, ext h⟩
+
+protected lemma continuous [topological_space Y] (f : locally_constant X Y) : continuous f :=
+f.is_locally_constant.continuous
+
+/-- The constant locally constant function on `X` with value `y : Y`. -/
+def const (X : Type*) {Y : Type*} [topological_space X] (y : Y) :
+  locally_constant X Y :=
+⟨function.const X y, is_locally_constant.const _⟩
+
+lemma range_finite [compact_space X] (f : locally_constant X Y) :
+  (set.range f).finite :=
+f.is_locally_constant.range_finite
+
+/-- Push forward of locally constant maps under any map, by post-composition. -/
+def map (f : Y → Z) : locally_constant X Y → locally_constant X Z :=
+λ g, ⟨f ∘ g, λ s, by { rw set.preimage_comp, apply g.is_locally_constant }⟩
+
+@[simp] lemma map_apply (f : Y → Z) (g : locally_constant X Y) : ⇑(map f g) = f ∘ g := rfl
+
+@[simp] lemma map_id : @map X Y Y _ id = id := by { ext, refl }
+
+@[simp] lemma map_comp {Y₁ Y₂ Y₃ : Type*} (g : Y₂ → Y₃) (f : Y₁ → Y₂) :
+  @map X _ _ _ g ∘ map f = map (g ∘ f) := by { ext, refl }
+
+section comap
+
+open_locale classical
+
+variables [topological_space Y]
+
+/-- Pull back of locally constant maps under any map, by pre-composition. -/
+noncomputable
+def comap (f : X → Y) :
+  locally_constant Y Z → locally_constant X Z :=
+if hf : continuous f
+then λ g, ⟨g ∘ f, λ s,
+  by { rw set.preimage_comp, apply hf.is_open_preimage, apply g.is_locally_constant }⟩
```

Comment:
Could you please add a lemma `is_locally_constant.comp_continuous` and use it here?

Reply:
Fixed.